File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String,Integer> hash = new HashMap<>();
        String answer = "";
        for (String p: participant){
            if (hash.containsKey(p)){
                hash.put(p,hash.get(p)+1);
            }
            else{
                hash.put(p,1);
            }
        }
        
        for (String c: completion){
            if (hash.get(c) == 1){
                hash.remove(c);
            }
            else{
                hash.put(c,hash.get(c)-1);
            }
        }
        
        for (String s: hash.keySet()){
            answer = s;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = dict()
    for p in participant:
        if p in dic:
            dic[p] += 1
        else:
            dic[p] = 1
    for c in completion:
        if dic[c] > 1:
            dic[c] -= 1
        else:
            del dic[c]
    
    
    answer = ''
    for key in dic:
        answer = key
    
    
    return answer